File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.cpp
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        for(int test_case = 1; test_case <= T; test_case++) {
            int n = sc.nextInt();
            int[][] arr = new int[n][n];
            for(int i = 0; i<n; i++) {
                for(int j = 0; j<n; j++) {
                    arr[i][j] = sc.nextInt();
                }
            }

            int[][] rotation_90 = rotate(arr, n);
            int[][] rotation_180 = rotate(rotation_90, n);
            int[][] rotation_270 = rotate(rotation_180, n);

            System.out.println("#" + test_case + " ");
            for(int i = 0; i<n; i++) {
                for(int j = 0; j<n; j++) {
                    System.out.print(rotation_90[i][j]);
                }
                System.out.print(" ");

                for(int j = 0; j<n; j++) {
                    System.out.print(rotation_180[i][j]);
                }
                System.out.print(" ");

                for(int j = 0; j<n; j++) {
                    System.out.print(rotation_270[i][j]);
                }
                System.out.println();
            }
        }
    }

    public static int[][] rotate(int[][] arr, int size) {
        int[][] result = new int[size][size];
        for(int i = 0; i<size; i++) {
            for(int j = 0; j<size; j++) {
                result[i][j] = arr[size - 1 - j][i];
            }
        }
        return result;
    }
}

File: SWEA/D2/1961. 숫자 배열 회전/숫자 배열 회전.py
T = int(input())

def rotate(arr, n):
    result = [[0 for _ in range(n)] for _ in range(n)]
    for i in range(n):
        for j in range(n):
            result[i][j] = arr[n - 1 - j][i]
    return result

for test_case in range(1, 11):
    n = int(input())
    arr = list(input().split() for _ in range(n))
    arr_90 = rotate(arr,n)
    arr_180 = rotate(arr_90,n)
    arr_270 = rotate(arr_180,n)
    
    print("#{}" .format(test_case))
    for i in range(n):
        for j in range(n):
            print(arr_90[i][j], end="")
        print(end=" ")
        for j in range(n):
            print(arr_180[i][j], end="")
        print(end=" ")
        for j in range(n):
            print(arr_270[i][j], end="")
        print()